client/web/src/search/navbarSearchQueryState.ts

```diff
@@ -5,21 +5,64 @@
 // (see https://github.com/sourcegraph/sourcegraph/issues/21200).
 import create from 'zustand'
 
-import { QueryState, SubmitSearchParameters, submitSearch } from './helpers'
+import { FilterType } from '@sourcegraph/shared/src/search/query/filters'
+import { appendFilter, updateFilter } from '@sourcegraph/shared/src/search/query/transformer'
+import { filterExists } from '@sourcegraph/shared/src/search/query/validate'
+
+import { QueryState, SubmitSearchParameters, submitSearch, toggleSearchFilter } from './helpers'
 
 type QueryStateUpdate = QueryState | ((queryState: QueryState) => QueryState)
 
+export type QueryUpdate =
+    | {
+          type: 'appendFilter'
+          field: FilterType
+          value: string
+          /**
+           * If true, the filter will only be appended a filter with the same name
+           * doesn't already exist in the query.
+           */
+          unique?: true
+      }
+    | {
+          type: 'updateOrAppendFilter'
+          field: FilterType
+          value: string
+      }
+    // Only exists for the filters from the serach sidebar since they come in
```

Comment:
```suggestion
    // Only exists for the filters from the search sidebar since they come in
```

I make this typo all the time :3